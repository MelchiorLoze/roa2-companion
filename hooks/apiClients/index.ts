export { useGameApiClient } from './useGameApiClient/useGameApiClient';
