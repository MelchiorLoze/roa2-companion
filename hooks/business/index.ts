export { useAuth } from './useAuth/useAuth';
export { useCurrencyBalance } from './useCurrencyBalance/useCurrencyBalance';
export { useLeaderboardStats } from './useLeaderboardStats/useLeaderboardStats';
export { useRotatingCoinShop } from './useRotatingCoinShop/useRotatingCoinShop';
export { useUserStats } from './useUserStats/useUserStats';
