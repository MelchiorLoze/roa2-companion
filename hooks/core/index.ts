export { useStorageState } from '../core/useStorageState/useStorageState';
export { useAppFonts } from './useAppFonts/useAppFonts';
export { useHttpClient } from './useHttpClient/useHttpClient';
