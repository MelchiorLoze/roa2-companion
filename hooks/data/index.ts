export { useGetEntityToken } from './useGetEntityToken/useGetEntityToken';
export { useGetInventoryItems } from './useGetInventoryItems/useGetInventoryItems';
export { useGetItems } from './useGetItems/useGetItems';
export { useGetPlayerStatistics } from './useGetPlayerStatistics/useGetPlayerStatistics';
export { useGetUserReadOnlyData } from './useGetUserReadOnlyData/useGetUserReadOnlyData';
export { useLoginWithEmail } from './useLoginWithEmail/useLoginWithEmail';
export { useGetMyRotationalCoinStore } from './useMyRotationalCoinStore/useGetMyRotationalCoinStore';
