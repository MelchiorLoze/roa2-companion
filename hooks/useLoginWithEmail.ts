import { BASE_URL, TITLE_ID } from '@/constants';
import { useMutation } from '@tanstack/react-query';

type LoginWithEmailAddressRequest = {
  email: string;
  password: string;
};

type LoginWithEmailAddressResponse = {
  EntityToken: string;
  TokenExpiration: string;
};

const loginWithEmailAddress = async ({ email, password }: LoginWithEmailAddressRequest) => {
  const response = await fetch(`${BASE_URL}/Client/LoginWithEmailAddress`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      TitleId: TITLE_ID,
      Email: email,
      Password: password,
    }),
  });

  if (!response.ok) {
    throw new Error('Failed to login');
  }

  return (await response.json()).data.EntityToken as LoginWithEmailAddressResponse;
};

export const useLoginWithEmail = () => {
  const {
    data,
    mutate: loginWithEmail,
    isError,
    isPending,
  } = useMutation({
    mutationFn: loginWithEmailAddress,
  });

  return { data, loginWithEmail, isLoading: isPending, isError };
};
