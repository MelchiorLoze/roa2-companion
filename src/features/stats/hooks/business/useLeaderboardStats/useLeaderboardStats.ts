import { type barDataItem, type lineDataItem } from 'react-native-gifted-charts';
import { useUnistyles } from 'react-native-unistyles';
import { type UnistylesTheme } from 'react-native-unistyles/lib/typescript/src/types';

import { useSeason } from '@/features/stats/contexts/SeasonContext/SeasonContext';
import { getRank, type LeaderboardEntry, MAX_AETHEREAN_PLAYERS, type Rank } from '@/types/rank';

import { useCommunityLeaderboard } from '../../data/useCommunityLeaderboard/useCommunityLeaderboard';

const DISTRIBUTION_PRECISION = 10;

const roundedElo = (elo: number) => Math.floor(elo / DISTRIBUTION_PRECISION) * DISTRIBUTION_PRECISION;

type RankDistributionBarDataItem = barDataItem;
const getRankDistribution = (
  leaderboardEntries: LeaderboardEntry[],
  theme: UnistylesTheme,
): RankDistributionBarDataItem[] => {
  const result = leaderboardEntries.reduce(
    (acc, entry) => {
      const rank = getRank(entry.elo, entry.position);
      if (acc[rank]?.value) acc[rank].value++;
      else acc[rank] = { value: 1, frontColor: theme.color[rank] };
      return acc;
    },
    {} as Record<Rank, RankDistributionBarDataItem>,
  );

  return Object.values(result);
};

type EloDistributionLineDataItem = { elo: number } & lineDataItem;
const getEloDistribution = (
  leaderboardEntries: LeaderboardEntry[],
  firstPlayerElo: number,
  lastPlayerElo: number,
  userElo: number,
): EloDistributionLineDataItem[] => {
  const totalEloRange = firstPlayerElo - lastPlayerElo + 1;
  const result = Array.from(
    { length: Math.ceil(totalEloRange / DISTRIBUTION_PRECISION) },
    (_, i) => i * DISTRIBUTION_PRECISION + roundedElo(lastPlayerElo),
  ).reduce(
    (acc, elo) => ({ ...acc, [elo]: { elo: elo, value: 0 } as EloDistributionLineDataItem }),
    {} as Record<number, EloDistributionLineDataItem>,
  );

  leaderboardEntries.forEach((entry) => {
    const elo = roundedElo(entry.elo);
    if (result[elo].value) result[elo].value++;
    else result[elo].value = 1;
  });

  const userEloRounded = roundedElo(userElo);
  if (result[userEloRounded]) result[userEloRounded].dataPointColor = 'red';

  return Object.values(result).sort((a, b) => a.elo - b.elo);
};

export const useLeaderboardStats = (userElo = 0) => {
  const { leaderboardId } = useSeason();
  const { leaderboardEntries, isLoading: isLoadingLeaderboard } = useCommunityLeaderboard(leaderboardId);

  const { theme } = useUnistyles();

  const firstPlayerElo = leaderboardEntries[0]?.elo ?? 0;
  const lastPlayerElo = Math.min(0, leaderboardEntries[leaderboardEntries.length - 1]?.elo ?? 0);
  const lastAethereanElo =
    leaderboardEntries.findLast((entry) => entry.elo > 1800 && entry.position < MAX_AETHEREAN_PLAYERS)?.elo ?? 0;

  return {
    firstPlayerElo,
    lastPlayerElo,
    lastAethereanElo,
    rankDistribution: getRankDistribution(leaderboardEntries, theme),
    eloDistribution: getEloDistribution(leaderboardEntries, firstPlayerElo, lastPlayerElo, userElo),
    isLoading: isLoadingLeaderboard,
  };
};
