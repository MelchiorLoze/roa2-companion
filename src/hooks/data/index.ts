// GAME API
export { useGetEntityToken } from './useGetEntityToken/useGetEntityToken';
export { useGetInventoryItems } from './useGetInventoryItems/useGetInventoryItems';
export { useGetLeaderboardAroundPlayer } from './useGetLeaderboardAroundPlayer/useGetLeaderboardAroundPlayer';
export { useGetPlayerStatistics } from './useGetPlayerStatistics/useGetPlayerStatistics';
export { useGetUserReadOnlyData } from './useGetUserReadOnlyData/useGetUserReadOnlyData';
export { useLoginWithEmail } from './useLoginWithEmail/useLoginWithEmail';
export { useSendAccountRecoveryEmail } from './useSendAccountRecoveryEmail/useSendAccountRecoveryEmail';

// STEAM API
export { useCommunityLeaderboard } from './useCommunityLeaderboard/useCommunityLeaderboard';
export { useCommunityLeaderboards } from './useCommunityLeaderboards/useCommunityLeaderboards';
