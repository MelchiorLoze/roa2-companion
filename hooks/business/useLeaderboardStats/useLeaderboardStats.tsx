import { useMemo } from 'react';
import { barDataItem, lineDataItem } from 'react-native-gifted-charts';
import { useUnistyles } from 'react-native-unistyles';

import { useCommunityLeaderboard, useCommunityLeaderboards } from '@/hooks/data';
import { getRank, Rank } from '@/types/rank';

import { useUserStats } from '../useUserStats/useUserStats';

const DISTRIBUTION_PRECISION = 10;

const roundedElo = (elo: number) => Math.floor(elo / DISTRIBUTION_PRECISION) * DISTRIBUTION_PRECISION;

export const useLeaderboardStats = () => {
  const { leaderboards, isLoading: isLoadingLeaderboards } = useCommunityLeaderboards();
  const { leaderboardEntries, isLoading: isLoadingLeaderboard } = useCommunityLeaderboard(leaderboards[1]?.id ?? -1);
  const { stats, isLoading: isLoadingUserStats } = useUserStats();

  const { theme } = useUnistyles();

  const firstPlayerElo = leaderboardEntries[0]?.elo ?? 0;
  const lastPlayerElo = Math.min(0, leaderboardEntries[leaderboardEntries.length - 1]?.elo ?? 0);
  const lastAethereanElo = leaderboardEntries.findLast((entry, index) => entry.elo > 1800 && index < 100)?.elo ?? 1800;

  const distributionPerRank = leaderboardEntries.reduce(
    (acc, entry) => {
      const rank = getRank(entry.elo, entry.position);
      if (acc[rank]?.value) acc[rank].value++;
      else acc[rank] = { value: 1, frontColor: theme.color[rank] };
      return acc;
    },
    {} as Record<Rank, barDataItem>,
  );

  const distributionPerElo = useMemo(() => {
    if (!leaderboardEntries.length || !firstPlayerElo) return [];

    const result = Array.from(
      { length: Math.ceil(firstPlayerElo / DISTRIBUTION_PRECISION) },
      (_, i) => i * DISTRIBUTION_PRECISION,
    ).reduce(
      (acc, elo) => {
        acc[elo] = { value: 0, label: elo.toString() };
        return acc;
      },
      {} as Record<number, lineDataItem>,
    );

    leaderboardEntries.forEach((entry) => {
      const elo = roundedElo(entry.elo);
      if (result[elo].value) result[elo].value++;
      else result[elo].value = 1;
    });

    if (stats?.rankedElo != null) {
      const playerElo = roundedElo(stats.rankedElo);
      if (result[playerElo]) result[playerElo].dataPointColor = 'red';
    }

    return result;
  }, [firstPlayerElo, leaderboardEntries, stats?.rankedElo]);

  return {
    firstPlayerElo,
    lastPlayerElo,
    lastAethereanElo,
    rankDistribution: Object.values(distributionPerRank),
    eloDistribution: Object.values(distributionPerElo),
    isLoading: isLoadingUserStats || isLoadingLeaderboards || isLoadingLeaderboard,
  };
};
