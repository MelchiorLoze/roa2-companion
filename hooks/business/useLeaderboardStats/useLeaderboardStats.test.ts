import { renderHook } from '@testing-library/react-native';

import { useCommunityLeaderboard, useCommunityLeaderboards } from '@/hooks/data';

import { useLeaderboardStats } from './useLeaderboardStats';

jest.mock('@/hooks/data');

const useCommunityLeaderboardsMock = jest.mocked(useCommunityLeaderboards);
const defaultCommunityLeaderboardsState: ReturnType<typeof useCommunityLeaderboards> = {
  leaderboards: [
    { id: 123, name: 'season_1', displayName: 'Season 1', entryCount: 100 },
    { id: 456, name: 'season_2', displayName: 'Season 2', entryCount: 200 },
    { id: 789, name: 'season_3', displayName: 'Season 3', entryCount: 300 },
  ],
  isLoading: false,
  isError: false,
};

const useCommunityLeaderboardMock = jest.mocked(useCommunityLeaderboard);
const defaultCommunityLeaderboardState: ReturnType<typeof useCommunityLeaderboard> = {
  leaderboardEntries: [],
  isLoading: false,
  isError: false,
};

const renderUseLeaderboardStats = (...props: Parameters<typeof useLeaderboardStats>) => {
  const { result } = renderHook(() => useLeaderboardStats(...props));

  expect(useCommunityLeaderboardsMock).toHaveBeenCalledTimes(1);
  expect(useCommunityLeaderboardMock).toHaveBeenCalledTimes(1);

  return { result };
};

describe('useLeaderboardStats', () => {
  beforeEach(() => {
    useCommunityLeaderboardsMock.mockReturnValue(defaultCommunityLeaderboardsState);
    useCommunityLeaderboardMock.mockReturnValue(defaultCommunityLeaderboardState);
  });

  it('returns correct values when leaderboards are loading', () => {
    useCommunityLeaderboardsMock.mockReturnValue({
      ...defaultCommunityLeaderboardsState,
      leaderboards: [],
      isLoading: true,
    });

    const { result } = renderUseLeaderboardStats(1000);

    expect(useCommunityLeaderboardMock).toHaveBeenCalledWith(undefined);

    expect(result.current.isLoading).toBe(true);
    expect(result.current.firstPlayerElo).toBe(0);
    expect(result.current.lastPlayerElo).toBe(0);
    expect(result.current.lastAethereanElo).toBe(0);
    expect(result.current.rankDistribution).toEqual([]);
    expect(result.current.eloDistribution).toEqual([{ value: 0, elo: 0 }]);
  });

  it('returns correct values when leaderboard entries are loading', () => {
    useCommunityLeaderboardMock.mockReturnValue({
      ...defaultCommunityLeaderboardState,
      isLoading: true,
    });

    const { result } = renderUseLeaderboardStats(1000);

    expect(result.current.isLoading).toBe(true);
    expect(result.current.firstPlayerElo).toBe(0);
    expect(result.current.lastPlayerElo).toBe(0);
    expect(result.current.lastAethereanElo).toBe(0);
    expect(result.current.rankDistribution).toEqual([]);
    expect(result.current.eloDistribution).toEqual([{ value: 0, elo: 0 }]);
  });

  it('fetches leaderboard from last leaderboards id', () => {
    renderUseLeaderboardStats(1000);

    expect(useCommunityLeaderboardMock).toHaveBeenCalledWith(789);
  });

  it('matches snapshot for complete for leaderboard', () => {
    useCommunityLeaderboardMock.mockReturnValue({
      ...defaultCommunityLeaderboardsState,
      leaderboardEntries: [
        // Aetherean
        { steamId: 1, position: 1, elo: 2162 },
        { steamId: 2, position: 2, elo: 2089 },
        { steamId: 3, position: 3, elo: 1910 },
        { steamId: 4, position: 4, elo: 1837 },
        // Grandmaster
        { steamId: 5, position: 5, elo: 1791 },
        // Master
        { steamId: 6, position: 6, elo: 1655 },
        { steamId: 7, position: 7, elo: 1649 },
        { steamId: 8, position: 8, elo: 1523 },
        // Diamond
        { steamId: 9, position: 9, elo: 1450 },
        { steamId: 10, position: 10, elo: 1300 },
        // Platinum
        { steamId: 11, position: 11, elo: 1258 },
        { steamId: 12, position: 12, elo: 1111 },
        // Gold
        { steamId: 13, position: 13, elo: 1000 },
        { steamId: 14, position: 14, elo: 920 },
        // Silver
        { steamId: 15, position: 15, elo: 820 },
        { steamId: 16, position: 16, elo: 752 },
        // Bronze
        { steamId: 17, position: 17, elo: 699 },
        { steamId: 18, position: 18, elo: 510 },
        // Stone
        { steamId: 19, position: 19, elo: 400 },
        { steamId: 20, position: 20, elo: 300 },
        { steamId: 25, position: 25, elo: -100 },
      ],
    });

    const { result } = renderUseLeaderboardStats(1000);

    expect(result.current).toMatchSnapshot();
  });

  it('returns 0 as lastPlayerElo when it is above 0', () => {
    useCommunityLeaderboardMock.mockReturnValue({
      ...defaultCommunityLeaderboardState,
      leaderboardEntries: [
        { steamId: 1, position: 1, elo: 1000 },
        { steamId: 2, position: 2, elo: 50 },
      ],
    });

    const { result } = renderUseLeaderboardStats(1000);

    expect(result.current.lastPlayerElo).toBe(0);
  });
});
