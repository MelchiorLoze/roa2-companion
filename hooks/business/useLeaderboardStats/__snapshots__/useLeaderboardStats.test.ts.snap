// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useLeaderboardStats matches snapshot for complete for leaderboard 1`] = `
{
  "eloDistribution": [
    {
      "elo": -100,
      "value": 1,
    },
    {
      "elo": -90,
      "value": 0,
    },
    {
      "elo": -80,
      "value": 0,
    },
    {
      "elo": -70,
      "value": 0,
    },
    {
      "elo": -60,
      "value": 0,
    },
    {
      "elo": -50,
      "value": 0,
    },
    {
      "elo": -40,
      "value": 0,
    },
    {
      "elo": -30,
      "value": 0,
    },
    {
      "elo": -20,
      "value": 0,
    },
    {
      "elo": -10,
      "value": 0,
    },
    {
      "elo": 0,
      "value": 0,
    },
    {
      "elo": 10,
      "value": 0,
    },
    {
      "elo": 20,
      "value": 0,
    },
    {
      "elo": 30,
      "value": 0,
    },
    {
      "elo": 40,
      "value": 0,
    },
    {
      "elo": 50,
      "value": 0,
    },
    {
      "elo": 60,
      "value": 0,
    },
    {
      "elo": 70,
      "value": 0,
    },
    {
      "elo": 80,
      "value": 0,
    },
    {
      "elo": 90,
      "value": 0,
    },
    {
      "elo": 100,
      "value": 0,
    },
    {
      "elo": 110,
      "value": 0,
    },
    {
      "elo": 120,
      "value": 0,
    },
    {
      "elo": 130,
      "value": 0,
    },
    {
      "elo": 140,
      "value": 0,
    },
    {
      "elo": 150,
      "value": 0,
    },
    {
      "elo": 160,
      "value": 0,
    },
    {
      "elo": 170,
      "value": 0,
    },
    {
      "elo": 180,
      "value": 0,
    },
    {
      "elo": 190,
      "value": 0,
    },
    {
      "elo": 200,
      "value": 0,
    },
    {
      "elo": 210,
      "value": 0,
    },
    {
      "elo": 220,
      "value": 0,
    },
    {
      "elo": 230,
      "value": 0,
    },
    {
      "elo": 240,
      "value": 0,
    },
    {
      "elo": 250,
      "value": 0,
    },
    {
      "elo": 260,
      "value": 0,
    },
    {
      "elo": 270,
      "value": 0,
    },
    {
      "elo": 280,
      "value": 0,
    },
    {
      "elo": 290,
      "value": 0,
    },
    {
      "elo": 300,
      "value": 1,
    },
    {
      "elo": 310,
      "value": 0,
    },
    {
      "elo": 320,
      "value": 0,
    },
    {
      "elo": 330,
      "value": 0,
    },
    {
      "elo": 340,
      "value": 0,
    },
    {
      "elo": 350,
      "value": 0,
    },
    {
      "elo": 360,
      "value": 0,
    },
    {
      "elo": 370,
      "value": 0,
    },
    {
      "elo": 380,
      "value": 0,
    },
    {
      "elo": 390,
      "value": 0,
    },
    {
      "elo": 400,
      "value": 1,
    },
    {
      "elo": 410,
      "value": 0,
    },
    {
      "elo": 420,
      "value": 0,
    },
    {
      "elo": 430,
      "value": 0,
    },
    {
      "elo": 440,
      "value": 0,
    },
    {
      "elo": 450,
      "value": 0,
    },
    {
      "elo": 460,
      "value": 0,
    },
    {
      "elo": 470,
      "value": 0,
    },
    {
      "elo": 480,
      "value": 0,
    },
    {
      "elo": 490,
      "value": 0,
    },
    {
      "elo": 500,
      "value": 0,
    },
    {
      "elo": 510,
      "value": 1,
    },
    {
      "elo": 520,
      "value": 0,
    },
    {
      "elo": 530,
      "value": 0,
    },
    {
      "elo": 540,
      "value": 0,
    },
    {
      "elo": 550,
      "value": 0,
    },
    {
      "elo": 560,
      "value": 0,
    },
    {
      "elo": 570,
      "value": 0,
    },
    {
      "elo": 580,
      "value": 0,
    },
    {
      "elo": 590,
      "value": 0,
    },
    {
      "elo": 600,
      "value": 0,
    },
    {
      "elo": 610,
      "value": 0,
    },
    {
      "elo": 620,
      "value": 0,
    },
    {
      "elo": 630,
      "value": 0,
    },
    {
      "elo": 640,
      "value": 0,
    },
    {
      "elo": 650,
      "value": 0,
    },
    {
      "elo": 660,
      "value": 0,
    },
    {
      "elo": 670,
      "value": 0,
    },
    {
      "elo": 680,
      "value": 0,
    },
    {
      "elo": 690,
      "value": 1,
    },
    {
      "elo": 700,
      "value": 0,
    },
    {
      "elo": 710,
      "value": 0,
    },
    {
      "elo": 720,
      "value": 0,
    },
    {
      "elo": 730,
      "value": 0,
    },
    {
      "elo": 740,
      "value": 0,
    },
    {
      "elo": 750,
      "value": 1,
    },
    {
      "elo": 760,
      "value": 0,
    },
    {
      "elo": 770,
      "value": 0,
    },
    {
      "elo": 780,
      "value": 0,
    },
    {
      "elo": 790,
      "value": 0,
    },
    {
      "elo": 800,
      "value": 0,
    },
    {
      "elo": 810,
      "value": 0,
    },
    {
      "elo": 820,
      "value": 1,
    },
    {
      "elo": 830,
      "value": 0,
    },
    {
      "elo": 840,
      "value": 0,
    },
    {
      "elo": 850,
      "value": 0,
    },
    {
      "elo": 860,
      "value": 0,
    },
    {
      "elo": 870,
      "value": 0,
    },
    {
      "elo": 880,
      "value": 0,
    },
    {
      "elo": 890,
      "value": 0,
    },
    {
      "elo": 900,
      "value": 0,
    },
    {
      "elo": 910,
      "value": 0,
    },
    {
      "elo": 920,
      "value": 1,
    },
    {
      "elo": 930,
      "value": 0,
    },
    {
      "elo": 940,
      "value": 0,
    },
    {
      "elo": 950,
      "value": 0,
    },
    {
      "elo": 960,
      "value": 0,
    },
    {
      "elo": 970,
      "value": 0,
    },
    {
      "elo": 980,
      "value": 0,
    },
    {
      "elo": 990,
      "value": 0,
    },
    {
      "dataPointColor": "red",
      "elo": 1000,
      "value": 1,
    },
    {
      "elo": 1010,
      "value": 0,
    },
    {
      "elo": 1020,
      "value": 0,
    },
    {
      "elo": 1030,
      "value": 0,
    },
    {
      "elo": 1040,
      "value": 0,
    },
    {
      "elo": 1050,
      "value": 0,
    },
    {
      "elo": 1060,
      "value": 0,
    },
    {
      "elo": 1070,
      "value": 0,
    },
    {
      "elo": 1080,
      "value": 0,
    },
    {
      "elo": 1090,
      "value": 0,
    },
    {
      "elo": 1100,
      "value": 0,
    },
    {
      "elo": 1110,
      "value": 1,
    },
    {
      "elo": 1120,
      "value": 0,
    },
    {
      "elo": 1130,
      "value": 0,
    },
    {
      "elo": 1140,
      "value": 0,
    },
    {
      "elo": 1150,
      "value": 0,
    },
    {
      "elo": 1160,
      "value": 0,
    },
    {
      "elo": 1170,
      "value": 0,
    },
    {
      "elo": 1180,
      "value": 0,
    },
    {
      "elo": 1190,
      "value": 0,
    },
    {
      "elo": 1200,
      "value": 0,
    },
    {
      "elo": 1210,
      "value": 0,
    },
    {
      "elo": 1220,
      "value": 0,
    },
    {
      "elo": 1230,
      "value": 0,
    },
    {
      "elo": 1240,
      "value": 0,
    },
    {
      "elo": 1250,
      "value": 1,
    },
    {
      "elo": 1260,
      "value": 0,
    },
    {
      "elo": 1270,
      "value": 0,
    },
    {
      "elo": 1280,
      "value": 0,
    },
    {
      "elo": 1290,
      "value": 0,
    },
    {
      "elo": 1300,
      "value": 1,
    },
    {
      "elo": 1310,
      "value": 0,
    },
    {
      "elo": 1320,
      "value": 0,
    },
    {
      "elo": 1330,
      "value": 0,
    },
    {
      "elo": 1340,
      "value": 0,
    },
    {
      "elo": 1350,
      "value": 0,
    },
    {
      "elo": 1360,
      "value": 0,
    },
    {
      "elo": 1370,
      "value": 0,
    },
    {
      "elo": 1380,
      "value": 0,
    },
    {
      "elo": 1390,
      "value": 0,
    },
    {
      "elo": 1400,
      "value": 0,
    },
    {
      "elo": 1410,
      "value": 0,
    },
    {
      "elo": 1420,
      "value": 0,
    },
    {
      "elo": 1430,
      "value": 0,
    },
    {
      "elo": 1440,
      "value": 0,
    },
    {
      "elo": 1450,
      "value": 1,
    },
    {
      "elo": 1460,
      "value": 0,
    },
    {
      "elo": 1470,
      "value": 0,
    },
    {
      "elo": 1480,
      "value": 0,
    },
    {
      "elo": 1490,
      "value": 0,
    },
    {
      "elo": 1500,
      "value": 0,
    },
    {
      "elo": 1510,
      "value": 0,
    },
    {
      "elo": 1520,
      "value": 1,
    },
    {
      "elo": 1530,
      "value": 0,
    },
    {
      "elo": 1540,
      "value": 0,
    },
    {
      "elo": 1550,
      "value": 0,
    },
    {
      "elo": 1560,
      "value": 0,
    },
    {
      "elo": 1570,
      "value": 0,
    },
    {
      "elo": 1580,
      "value": 0,
    },
    {
      "elo": 1590,
      "value": 0,
    },
    {
      "elo": 1600,
      "value": 0,
    },
    {
      "elo": 1610,
      "value": 0,
    },
    {
      "elo": 1620,
      "value": 0,
    },
    {
      "elo": 1630,
      "value": 0,
    },
    {
      "elo": 1640,
      "value": 1,
    },
    {
      "elo": 1650,
      "value": 1,
    },
    {
      "elo": 1660,
      "value": 0,
    },
    {
      "elo": 1670,
      "value": 0,
    },
    {
      "elo": 1680,
      "value": 0,
    },
    {
      "elo": 1690,
      "value": 0,
    },
    {
      "elo": 1700,
      "value": 0,
    },
    {
      "elo": 1710,
      "value": 0,
    },
    {
      "elo": 1720,
      "value": 0,
    },
    {
      "elo": 1730,
      "value": 0,
    },
    {
      "elo": 1740,
      "value": 0,
    },
    {
      "elo": 1750,
      "value": 0,
    },
    {
      "elo": 1760,
      "value": 0,
    },
    {
      "elo": 1770,
      "value": 0,
    },
    {
      "elo": 1780,
      "value": 0,
    },
    {
      "elo": 1790,
      "value": 1,
    },
    {
      "elo": 1800,
      "value": 0,
    },
    {
      "elo": 1810,
      "value": 0,
    },
    {
      "elo": 1820,
      "value": 0,
    },
    {
      "elo": 1830,
      "value": 1,
    },
    {
      "elo": 1840,
      "value": 0,
    },
    {
      "elo": 1850,
      "value": 0,
    },
    {
      "elo": 1860,
      "value": 0,
    },
    {
      "elo": 1870,
      "value": 0,
    },
    {
      "elo": 1880,
      "value": 0,
    },
    {
      "elo": 1890,
      "value": 0,
    },
    {
      "elo": 1900,
      "value": 0,
    },
    {
      "elo": 1910,
      "value": 1,
    },
    {
      "elo": 1920,
      "value": 0,
    },
    {
      "elo": 1930,
      "value": 0,
    },
    {
      "elo": 1940,
      "value": 0,
    },
    {
      "elo": 1950,
      "value": 0,
    },
    {
      "elo": 1960,
      "value": 0,
    },
    {
      "elo": 1970,
      "value": 0,
    },
    {
      "elo": 1980,
      "value": 0,
    },
    {
      "elo": 1990,
      "value": 0,
    },
    {
      "elo": 2000,
      "value": 0,
    },
    {
      "elo": 2010,
      "value": 0,
    },
    {
      "elo": 2020,
      "value": 0,
    },
    {
      "elo": 2030,
      "value": 0,
    },
    {
      "elo": 2040,
      "value": 0,
    },
    {
      "elo": 2050,
      "value": 0,
    },
    {
      "elo": 2060,
      "value": 0,
    },
    {
      "elo": 2070,
      "value": 0,
    },
    {
      "elo": 2080,
      "value": 1,
    },
    {
      "elo": 2090,
      "value": 0,
    },
    {
      "elo": 2100,
      "value": 0,
    },
    {
      "elo": 2110,
      "value": 0,
    },
    {
      "elo": 2120,
      "value": 0,
    },
    {
      "elo": 2130,
      "value": 0,
    },
    {
      "elo": 2140,
      "value": 0,
    },
    {
      "elo": 2150,
      "value": 0,
    },
    {
      "elo": 2160,
      "value": 1,
    },
  ],
  "firstPlayerElo": 2162,
  "isLoading": false,
  "lastAethereanElo": 1837,
  "lastPlayerElo": -100,
  "rankDistribution": [
    {
      "frontColor": "#BD7CC5",
      "value": 4,
    },
    {
      "frontColor": "#FC7575",
      "value": 1,
    },
    {
      "frontColor": "#9CE7AC",
      "value": 3,
    },
    {
      "frontColor": "#95DFEC",
      "value": 2,
    },
    {
      "frontColor": "#C3C4FF",
      "value": 2,
    },
    {
      "frontColor": "#FFC64E",
      "value": 2,
    },
    {
      "frontColor": "#C8C8D5",
      "value": 2,
    },
    {
      "frontColor": "#B87A2E",
      "value": 2,
    },
    {
      "frontColor": "#7F7A66",
      "value": 3,
    },
  ],
}
`;
